diffrax/nonlinear_solver/newton.py

```diff
@@ -81,8 +81,8 @@ class NewtonNonlinearSolver(AbstractNonlinearSolver):
     tolerate_nonconvergence: bool = False
 
     def __post_init__(self):
-        if self.max_steps is not None:
-            error_if(self.max_steps < 2, "max_steps must be at least 2.")
+        if self.max_steps and self.max_steps < 2:
```

Comment:
This should be `if self.max_steps is not None and self.max_steps < 2`. With this PR, `max_steps==0` will not raise an error when it should.

To avoid exactly these kinds of errors, I recommend avoiding implicit casts to bool.

Reply:
sounds good. applied.